lib/Sema/ConstraintSystem.cpp

```diff
@@ -497,6 +509,25 @@ ConstraintSystem::getCalleeLocator(ConstraintLocator *locator,
   return getConstraintLocator(anchor);
 }
 
+bool ConstraintSystem::isCallableNominalType(Type ty) {
```

Comment:
This kind of feels like it it belongs on the type (or even a nominal declaration) itself more than a part of constraint system interface...

Reply:
@xedin My original thought was to put it on the constraint system to benefit from its cache of member lookups, but I guess we just make a request for it instead. I've gone ahead and updated things so that `isCallableNominalType` is now a member of `TypeBase`, does that look good?

Thanks for the review!

Reply:
Looks great, thank you!

Reply:
Looks great, thank you!